@xen-orchestra/web-core/lib/components/CardNumbers.vue

```diff
@@ -0,0 +1,86 @@
+<!-- v1.0 -->
+<template>
+  <div class="card-numbers" :class="size">
+    <span class="label typo" :class="labelFontClass">{{ label }}</span>
+    <div class="values">
+      <span v-if="size === 'small' && max" class="value typo c2-semi-bold">{{ valueAsPercentage }}%</span>
+      <div class="value typo" :class="valueFontClass">
+        {{ value }}<span class="unit typo" :class="unitFontClass">{{ unit }}</span>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { computed } from 'vue'
+
+interface CardNumbersProps {
+  label: string
+  value: number
+  unit?: string
+  size: 'small' | 'medium'
+  max?: number
+}
```

Comment:
Since `max` will have no effect when `size` in not `small`, I'd suggest to update the code to prevent using `size="medium"` + `max`.

(I also moved required props together)

```suggestion
<script setup lang="ts" generic="TSize extends 'small' | 'medium'">
import { computed } from 'vue'

interface CardNumbersProps {
  label: string
  value: number
  size: TSize
  unit?: string
  max?: TSize extends 'small' ? number : never
}
```